src/typechecker/FStar.TypeChecker.TcInductive.fs

```diff
@@ -1248,7 +1248,7 @@ let mk_discriminator_and_indexed_projectors iquals                   (* Qualifie
                   else pos (Pat_wild (S.gen_bv (string_of_id x.ppname) None tun)), b)
               in
               let pat = pos (S.Pat_cons (S.lid_as_fv lid delta_constant (Some fvq), arg_pats)), None, S.bv_to_name projection in
-              let body = mk (Tm_match(arg_exp, [U.branch pat])) p in
+              let body = mk (Tm_match(arg_exp, None, [U.branch pat])) p in
```

Comment:
I wonder if by using a returns annotation, we can avoid the problems we have in typechecking some projectors with SMT

Reply:
Yeah, could try to see later.